modules/p2/ding_interests_by_history/views/views_handler_filter_interests_by_history.inc

```diff
@@ -0,0 +1,293 @@
+ * Definition of views_handler_filter_equality.
+ */
+
+/**
+ * Simple filter to handle equal to / not equal to filters
+ *
+ * @ingroup views_filter_handlers
+ */
+class Views_handler_filter_interests_by_history extends views_handler_filter_in_operator {
+  public $value_form_type = 'checkboxes';
+
+  // Stores all operations which are available on the form.
+  public $value_options = NULL;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function init(&$view, &$options) {
+    parent::init($view, $options);
+
+    $this->operator = $this->options['operator'];
+    $this->value = ding_interests_by_history_suggested_terms();
+    $this->group_info = $this->options['group_info']['default_group'];
+
+    // Compatibility: The new UI changed several settings.
+    if (!empty($options['exposed']) && !empty($options['expose']['optional']) && !isset($options['expose']['required'])) {
+      $this->options['expose']['required'] = !$options['expose']['optional'];
+    }
+    if (!empty($options['exposed']) && !empty($options['expose']['single']) && !isset($options['expose']['multiple'])) {
+      $this->options['expose']['multiple'] = !$options['expose']['single'];
+    }
+    if (!empty($options['exposed']) && !empty($options['expose']['operator']) && !isset($options['expose']['operator_id'])) {
+      $this->options['expose']['operator_id'] = $options['expose']['operator_id'] = $options['expose']['operator'];
+    }
+
+    if ($this->multiple_exposed_input()) {
+      $this->group_info = array_filter($options['group_info']['default_group_multiple']);
+      $this->options['expose']['multiple'] = TRUE;
+    }
+
+    // If there are relationships in the view, allow empty should be true
+    // so that we can do IS NULL checks on items. Not all filters respect
+    // allow empty, but string and numeric do and that covers enough.
+    if ($this->view->display_handler->get_option('relationships')) {
+      $this->definition['allow empty'] = TRUE;
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function construct() {
+    parent::construct();
+    $this->value_title = t('Options');
+    $this->value_options = NULL;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function get_value_options() {
+    if (isset($this->value_options)) {
+      return;
+    }
+
+    if (isset($this->definition['options callback']) && is_callable($this->definition['options callback'])) {
+      if (isset($this->definition['options arguments']) && is_array($this->definition['options arguments'])) {
+        $this->value_options = call_user_func_array($this->definition['options callback'], $this->definition['options arguments']);
+      }
+      else {
+        $this->value_options = call_user_func($this->definition['options callback']);
+      }
+    }
+    else {
+      $this->value_options = array(t('Yes'), t('No'));
+    }
+
+    return $this->value_options;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function expose_options() {
+    parent::expose_options();
```

Comment:
If a method in a subclass does nothing but call the same method in a parent class there is no reason to define it in the subclass.

This goes for `expose_form` as well.